### DISCUSSION TITLE
New and noob about grafana agent

### DISCUSSION THREAD

**Discussion by waardd** [grafana-cold-storage/agent#2717] [General]
Just reading into this grafana agent.
But is it a replacement optioon for node_exporter and promtail?
Im curtrently using that combination with prometheus/loki/grafana and im wondering if i could/would/must switch things around.
Not sure on how this is working and if its only for cloud solutions.
I have it all on prem.

**Comment by nabadger** [comment]
I'm also struggling to find documentation on what we should if we already run a dedicated promtail instance to ship loki logs.

If we need grafana-agent (for say tempo traces), do we get any benefit from using its built in promtail to ship loki logs? It sounds like it would be a simpler setup (consolidating on deployments), but what is the impact on user-experience/features?

**Comment by tpaschalis** [comment]
Hey there! 👋  Let me try to provide some context.

First up for metrics; the Agent makes use Prometheus' battle-tested code, but focuses only on a subset of its functionality, namely service discovery, scraping and remote_write.

That enables users to make a tradeoff of using smaller deployments of Agents for just scraping and sending out metrics (providing easier HA and sharding), and then transferring the responsibility of storing, querying, alerting and displaying these metrics data to a different system, either that is your on-prem centralized Prometheus, any hosted remote_write-compatible system or Grafana Cloud. 

So instead of scaling up an enormous sharded Prometheus to take care of an entire organization's infrastructure, the Agent can help split the load between smaller and more manageable deployments.

For metrics, the Agent also provides various [scaling mechanisms](https://grafana.com/docs/agent/latest/operation-guide/#horizontal-scaling), a [Kubernetes Operator](https://grafana.com/docs/agent/latest/operator/) as well as a multitude of bundled-in [integrations](https://grafana.com/docs/agent/latest/configuration/integrations/) so you don't have to run separate instances of various exporters.

> Im curtrently using that combination with prometheus/loki/grafana and im wondering if i could/would/must switch things around.
Not sure on how this is working and if its only for cloud solutions.

So, to answer your question, using the Agent for logs and metrics will provide the same experience, as it aims to be a drop-in replacement for Prometheus' scraping and remote_write, and would work the same on your on-prem environment or a cloud one.

What the Agent does, is enabling you to consolidate your deployments and choosing a different architecture to fit any needs.

**Comment by tpaschalis** [comment]
> But is it a replacement optioon for [...] promtail?

> I'm also struggling to find documentation on what we should if we already run a dedicated promtail instance to ship loki logs.
> If we need grafana-agent (for say tempo traces), do we get any benefit from using its built in promtail to ship loki logs? It sounds like it would be a simpler setup (consolidating on deployments), but what is the impact on user-experience/features?


cc @nabadger Same goes for logs. The Agent makes use of Promtail's code for collecting and shipping log entries to Loki; there should not be any impact on user-experience as we're keeping up with any new Promtail features with frequent updates and even have compatible configuration.

The main benefit is around allowing a more flexible architecture, making use of the various deployment modes that we support, as well trying out new ways to run and debug your logging pipelines such as [Grafana Agent Flow](https://grafana.com/docs/agent/next/flow/).

**Comment by waardd** [comment]
When i want to implement the agent.
How do i configure it?
I see the binary, but is there like a yml file to use like for example with node_exporter?

**Reply by akselleirv** [reply]
Hi @waardd,
The node exporter configuration can be found here: https://grafana.com/docs/agent/latest/configuration/integrations/node-exporter-config/

**Reply by tpaschalis** [reply]
Apologies for the belateed response. Yes, configuring the Agent is done through a config file and passing the `-config.file /path/to/cfg.yaml` command-line flag. The structure of the config file is described [here](https://grafana.com/docs/agent/latest/configuration/create-config-file/); the rest of the config pages cover specific subsets of it.
